cmd/tracee-ebpf/internal/config/config.go

```diff
@@ -0,0 +1,80 @@
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+type TraceeEbpfCliConfig struct {
+	PerfRingBuffers struct {
+		EventSubmitBufferSize int
+		BlobBufferSize        int
+	}
+	VerboseDebug bool
+	InstallPath  string
+	BuildPolicy  string
+	Trace        []string
+	Output       []string
+	Capture      []string
+}
+
+var Config TraceeEbpfCliConfig
+
+func LoadConfig(fileName string) error {
+	cfgFile, err := os.Open(fileName)
+
+	if err != nil {
+		return err
+	}
+
+	defer cfgFile.Close()
+
+	byteValue, err := ioutil.ReadAll(cfgFile)
+
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(byteValue, &Config)
+
+	return err
+}
+
+func DefaultConfigLocation() string {
```

Comment:
most importantly we need to be able to provide the config as a cli flag: `tracee-ebpf --config /path/to/file`

Reply:
I'm not sure how this relate to L276

```
app.Before = altsrc.InitInputSourceWithContext(app.Flags, altsrc.NewJSONSourceFromFlagFunc("config"))
```

Reply:
> most importantly we need to be able to provide the config as a cli flag: `tracee-ebpf --config /path/to/file`

I'll see if I can make it work with the `flag` package since from the `cli` it won't work because it parses after we need the location.

Reply:
@itaysk 

After some testing with the `flag` package, it seems to me this feature can't really work well with the cli package used (`urfave/cli`):
1. The package doesn't seem to play well with another flag package (i've attempted to do so with the standard `flag` package), each will complain about the other flags not being defined
2. If we define the config flag using `urfave/cli` it will be parsed alongside the other flags, but the config flag must be loaded before the other flags so the config can be loaded and then allow the cli flags to override it. There isn't really a way to tell the difference between an undefined flag and an undefined config field so we must have the config as the flag's default value, but then we would have to load the `config` flag before to find the config location before the rest of the flags, which is isn't possible in this package.

I think we should stick to the environment variable, it's a simple and easy solution, and if running from a container wouldn't be that different from passing a `--config` flag (you'd probably pass the config in a volume regardless so only the way you declare the config file location changes when running the `docker` command)